src/CPK.ts

```diff
@@ -130,64 +159,134 @@ class CPK {
+    return this.#ethLibAdapter?.getBalance(address)
   }
 
-  get isConnectedToSafe(): boolean {
-    return this.#isConnectedToSafe
+  /**
+   * Returns the ID of the connected network.
+   *
+   * @returns The ID of the connected network
+   */
+  async getNetworkId(): Promise<number | undefined> {
+    if (this.#safeAppsSdkConnector?.isSafeApp) {
+      const networkName = (await this.#safeAppsSdkConnector.getSafeInfo()).network
+      return getNetworkIdFromName(networkName)
+    }
+    if (!this.#ethLibAdapter) {
+      throw new Error('CPK ethLibAdapter uninitialized')
+    }
+    return this.#ethLibAdapter.getNetworkId()
   }
 
-  get contract(): Contract | undefined {
-    return this.#contractManager?.contract
+  /**
+   * Returns the safeAppsSdkConnector used by the CPK.
+   *
+   * @returns The safeAppsSdkConnector used by the CPK
+   */
+  get safeAppsSdkConnector(): SafeAppsSdkConnector | undefined {
+    return this.#safeAppsSdkConnector
   }
 
-  get multiSend(): Contract | undefined {
-    return this.#contractManager?.multiSend
+  /**
+   * Returns the contractManager used by the CPK.
+   *
+   * @returns The contractManager used by the CPK
+   */
+  get contractManager(): ContractManager | undefined {
+    return this.#contractManager
   }
 
-  get proxyFactory(): Contract | undefined {
-    return this.#contractManager?.proxyFactory
+  /**
+   * Returns the ethLibAdapter used by the CPK.
+   *
+   * @returns The ethLibAdapter used by the CPK
+   */
+  get ethLibAdapter(): EthLibAdapter | undefined {
+    return this.#ethLibAdapter
   }
 
-  get masterCopyAddress(): Address | undefined {
-    return this.#contractManager?.masterCopyAddress
+  /**
+   * Returns a list of the contract addresses which drive the CPK per network by network ID.
+   *
+   * @returns The list of the contract addresses which drive the CPK per network by network ID
+   */
+  get networks(): NormalizedNetworksConfig {
+    return this.#networks
   }
 
-  get fallbackHandlerAddress(): Address | undefined {
-    return this.#contractManager?.fallbackHandlerAddress
+  /**
+   * Checks if the CPK is connected to a Safe account or not.
+   *
+   * @returns TRUE if the CPK is connected to a Safe account
+   */
+  get isConnectedToSafe(): boolean {
+    return this.#isConnectedToSafe
   }
 
+  /**
+   * Returns the salt nonce used to deploy the Proxy Contract.
+   *
+   * @returns The salt nonce used to deploy the Proxy Contract
+   */
   get saltNonce(): string {
     return this.#saltNonce
   }
 
-  get address(): Address | undefined {
-    if (this.isSafeApp()) {
-      return this.#safeAppsSdkConnector.safeAppInfo?.safeAddress
+  /**
+   * Returns the address of the Proxy contract.
+   *
+   * @returns The address of the Proxy contract
+   */
+  get address(): Promise<Address | undefined> {
+    if (this.#safeAppsSdkConnector?.isSafeApp) {
+      return (async () => this.#safeAppsSdkConnector && (await this.#safeAppsSdkConnector.getSafeInfo()).safeAddress)()
```

Comment:
I'm definitely lacking knowledge on the architecture of cpk and stuff, but is it possible to cache the address when connecting?